File: API model/spoonacularTest.js
const API_KEY = "19f3af0561424a42884bb060ec999982";
const BASE_URL = "https://api.spoonacular.com/recipes/findByIngredients";

async function findRecipes(pantry) {
  const params = new URLSearchParams({
    ingredients: pantry.join(","),
    number: 5,
    ranking: 1,
    ignorePantry: true,
    apiKey: API_KEY
  });

  const res = await fetch(`${BASE_URL}?${params}`);
  if (!res.ok) throw new Error("Request failed");
  return res.json();
}

const pantry = ["pasta", "tomato", "garlic", "olive oil"];

findRecipes(pantry)
  .then(recipes => {
    recipes.forEach(r => {
      console.log(r.title);
      console.log("Used:", r.usedIngredients.map(i => i.name));
      console.log("Missing:", r.missedIngredients.map(i => i.name));
      console.log("----");
    });
  })
  .catch(console.error);

File: API model/spoonacular_test.py
import requests

SPOONACULAR_API_KEY = "19f3af0561424a42884bb060ec999982"
URL = "https://api.spoonacular.com/recipes/findByIngredients"

def find_recipes(pantry_items):
    params = {
        "ingredients": ",".join(pantry_items),
        "number": 5,
        "ranking": 1,
        "ignorePantry": True,
        "apiKey": SPOONACULAR_API_KEY
    }

    response = requests.get(URL, params=params)
    response.raise_for_status()
    return response.json()

pantry = ["chicken", "garlic", "rice", "olive oil"]
recipes = find_recipes(pantry)

for r in recipes:
    print(r["title"])
    print("Used:", [i["name"] for i in r["usedIngredients"]])
    print("Missing:", [i["name"] for i in r["missedIngredients"]])
    print("----")
